### PR TITLE
Add Zammad Ticketsystem to catalog

### PR DESCRIPTION
**Pull request by monotek**

### CODE REVIEW

**Comment by monotek** [comment]
Can somebody help me please?
Can't figure out whats failing:

```
=================================== FAILURES ===================================
71s
324
________________________________ test_versions _________________________________
71s
326
templates = {'resourceType': u'template', 'data': [{'files': None, 'upgradeFrom': u'', 'description': u'A highly-available in-memo...emplates/library:sysdig-cloud:0'}, 'license': u'', 'output': {'url': u''}, 'upgradeVersionLinks': None, 'labels': {}}]}
71s
328
    def test_versions(templates):
71s
329
        for template in templates:
71s
330
            # default version must be defined
71s
331
>           assert template.defaultVersion
71s
332
E           assert ''
71s
333
E            +  where '' = {'files': None, 'upgradeFrom': u'', 'description': u'Zammad is a web based open source helpdesk/customer support syste...emplates/library:zammad:0'}, 'license': u'AGPL-3.0', 'output': {'url': u''}, 'upgradeVersionLinks': None, 'labels': {}}.defaultVersion
71s
335
test_catalog.py:107: AssertionError
71s
336
====================== 1 failed, 4 passed in 5.00 seconds ======================
71s
337
ERROR: InvocationError: '/scratch/integration/.tox/py27/bin/py.test --duration=20'
71s
338
___________________________________ summary ____________________________________
71s
339
  flake8: commands succeeded
71s
340
ERROR:   py27: commands failed
```

**Comment by Ilya-Kuchaev** [comment]
+1

**Comment by LLParse** [comment]
@monotek There's a bug in rancher-catalog-service's usage of yaml parser. The value of your `version` parameter is parsed as a float64 but treated as a string. I'll open a bug for this.

Temporary workaround is to put quotes around your `version` token in config.yml

File: templates/zammad/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+version: '2'
+
+services:
+
```
Comment (rawmind0) [inline templates/zammad/0/docker-compose.yml]: # Remove elasticsearch service definition

File: templates/zammad/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      io.rancher.container.start_once: 'true'
+    restart: on-failure
+    volumes:
+      - data-zammad:/home/zammad
+
+  zammad-railsserver:
+    image: zammad/zammad-docker-compose:zammad-railsserver
+    labels:
+      io.rancher.container.pull_image: always
+    links:
```
Comment (rawmind0) [inline templates/zammad/0/docker-compose.yml]: # Set link based in elasticsearch service
--      - elasticsearch
++      - ${zammad_elasticsearch_service}: elasticsearch

File: templates/zammad/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      - elasticsearch
+      - postgresql
+      - zammad
+    depends_on:
```
Comment (rawmind0) [inline templates/zammad/0/docker-compose.yml]: # Remove elasticsearch dependency
--      - elasticsearch

File: templates/zammad/0/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      - elasticsearch
+      - postgresql
+      - zammad
+    restart: always
+
+  zammad-scheduler:
+    image: zammad/zammad-docker-compose:zammad-scheduler
+    labels:
+      io.rancher.container.pull_image: always
+    links:
```
Comment (rawmind0) [inline templates/zammad/0/docker-compose.yml]: # Set link based in elasticsearch service
--      - elasticsearch
++      - ${zammad_elasticsearch_service}: elasticsearch

File: templates/zammad/0/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      - elasticsearch
+      - postgresql
+      - zammad
+    depends_on:
```
Comment (rawmind0) [inline templates/zammad/0/docker-compose.yml]: # Remove elasticsearch dependency
--      - elasticsearch

File: templates/zammad/0/rancher-compose.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+.catalog:
+  name: "Zammad"
+  version: "1.2"
+  description: "Zammad is a web based open source helpdesk/customer support system"
+  uuid: zammad-1
+  minimum_rancher_version: v1.2.0
+  questions:
```
Comment (rawmind0) [inline templates/zammad/0/rancher-compose.yml]: # Add service combo question to select elasticsearch service. You need to deploy it before.
    - variable: "zammad_elasticsearch_service"
      description: |
        Choose the elasticsearch service to use. 
      label: "Elasticsearch stack/service"
      default: "elasticsearch-2/kopf"
      required: true
      type: "service"

File: templates/zammad/0/rancher-compose.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (rawmind0) [inline templates/zammad/0/rancher-compose.yml]: # Remove elasticsearch scale definition

**Comment by rawmind0** [comment]
Hi @monotek .. Thanks so much for your cool PR and contribute....

Have you consider to use the elasticsearch package instead add it inside your package?? You could deploy it and link your service with it, and take profit of the already created packaged. What do you think??

Best regards....

**Comment by monotek** [comment]
We use an own elasticsearch image because Zammad needs the mapper-attachments plugin to work, which is installed in the new container image.

See:  https://github.com/zammad/zammad-docker-compose/blob/master/containers/elasticsearch/Dockerfile

**Comment by rawmind0** [comment]
If you need an specific plugin in your elasticsearch, ok.....I've removed changes request....

Tested and working ok.... LGTM....Thanks....

**Pull request merged**